ddev/ddev
# fix: add correct mimetype for xsl and xslt, fixes #5457

Add correct mimetype for xsl and xslt

Resolves: #5457

## The Issue
XSL and XSLT files are being served by nginx with the default mimetype (application/octet-stream instead of [the standard](https://www.w3.org/TR/xslt20/#media-type-registration) `application/xslt+xml`.

This means that the `content-type` header is sending the wrong application type to the XSLT parser/transformer application and it therefore fails to process. 

## How This PR Solves The Issue
This code change ~~overrides the shipping debian /etc/nginx/mime.types with our own ddev-managed /etc/nginx/mime.types file so we can more easily make fixes to the mime types while waiting for the upstream distro to be updated.~~ adds a `types{}` section to the nginx.conf just under the `include mime.types` directive where we can add custom mimetypes for ddev.

## Manual Testing Instructions
1. Build your ddev instance
2. Put an XSL file such as /test.xsl in the document root. 
3. `curl -is "https://yoursite.ddev.site/test.xsl" | grep content-type`
You should see `content-type: application/xslt+xml` in the output.

## Automated Testing Overview
No tests added. Is there a framework for scripting a test for this in ddev? I read through [the testing docs](https://github.com/ddev/ddev/blob/master/docs/content/developers/building-contributing.md#testing) but that all surrounds tests for ddev itself. Please let me know if there is a recommended way I can create a test for this.

## Related Issue Link(s)
https://github.com/ddev/ddev/issues/5457

## Release/Deployment Notes
This does not affect the behavior of other parts of ddev itself.

If a user's application relies on a non-standards-compliant content-type mimetype for .xsl / .xslt files, their application may fail, as we are now serving .xsl and .xslt files according to W3 standards.

This merge request sponsored by [Chromatic](https://chromatichq.com/).
